.github/workflows/presto-release-publish.yml

```
@@ -0,0 +1,449 @@
+        description: 'Publish native docker images'
+        type: boolean
+        default: true
+        required: false
+      tag_image_as_latest:
+        description: 'Tag the image as latest'
+        type: boolean
+        default: true
+        required: false
+      dependency_image:
+        description: 'Dependency image(e.g., prestodb/presto-native-dependency:0.290-20241014120930-e1fc090)'
+        required: false
+        default: ''
+      publish_docs:
+        description: 'Publish docs'
+        type: boolean
+        default: true
+        required: false
+
+env:
+  DOCKER_REPO: ${{ github.repository }}
+  ORG_NAME: ${{ github.repository_owner }}
+  IMAGE_NAME: presto-native
+  RELEASE_BRANCH: release-${{ inputs.RELEASE_VERSION }}
+  RELEASE_TAG: ${{ inputs.RELEASE_VERSION }}
+  RELEASE_NOTES_COMMIT: ${{ inputs.release-notes-commit }}
+
+jobs:
+  tag-release:
+    if: ${{ github.event.inputs.publish_release_tag == 'true' }}
+    runs-on: ubuntu-latest
+    environment: release
+    steps:
+      - name: Validate release notes commit
+        if: ${{ github.event.inputs.release-notes-commit == '' }}
+        run: |
+          echo "Error: release-notes-commit is required when publish_release_tag is true"
+          exit 1
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ env.RELEASE_BRANCH }}
+          token: ${{ secrets.PRESTODB_CI_TOKEN }}
+          fetch-depth: 0
+          fetch-tags: true
+          show-progress: false
+
+      - name: Configure Git
+        run: |
+          git config --global user.email "ci@lists.prestodb.io"
+          git config --global user.name "prestodb-ci"
+          git config pull.rebase false
+
+      - name: Cherry-pick release notes
+        run: |
+          git cherry-pick ${{ env.RELEASE_NOTES_COMMIT }}
+
+      - name: Delete existing release tag
+        run: |
+          git push origin :${{ env.RELEASE_TAG }} || true
```

Can we use
```
git push --delete origin ${{ env.RELEASE_TAG }} || true
```
to be a bit more reader friendly?

The || true will always result in success of the step. It is no problem if the deletion of the tag fails?

---

Add `|| true` to avoid error when the tag deleted by accident, for example, deleted before re-run the job. In this case, the error can be ignored